### PR TITLE
Remove Reliance on Wikipedia Sections

### PR DESCRIPTION
**Pull request by bkb2135**

### CODE REVIEW

File: prompting/tools/datasets/wiki.py [hunk 1]
```diff
@@ -77,13 +72,32 @@ def _get_random_titles(pages=10, seed=42) -> List:
...
+        section_content_tag = section.find_next('p')
+        if section_content_tag is not None:
+            section_content = section_content_tag.text
+            sections[section_title] = section_content
+
+    return sections
 
 def process_page(
-    page, valid_header: callable = None, valid_content: callable = None
+    page, exclude_sections: list = None, valid_section: callable = None, selector: Selector = None
```
Comment (Hollyqui) [inline prompting/tools/datasets/wiki.py:100]: Best to type as Optional[list] or list | None if you're allowing values to be nullable

**Review by Hollyqui — APPROVED** [review]
LGTM, check comments

File: prompting/utils/logging.py [hunk 2]
```diff
@@ -103,7 +106,7 @@ def reinit_wandb(self):
 
 def log_event(self, event):
     if not self.config.neuron.dont_save_events:
-        logger.log("EVENTS", "events", **event)
+        logger.log(38, event)
```
Comment (dbobrenko) [inline prompting/utils/logging.py:109]: What does 38 mean?

File: prompting/tools/datasets/wiki.py [hunk 3]
```diff
@@ -177,7 +182,7 @@ def get(
         name: str,
         selector: Selector = None,
         include: List = None,
-        exclude: List = None,
+        exclude: List = [],
```
Comment (dbobrenko) [inline prompting/tools/datasets/wiki.py:185]: `None `default type would be better for lists/dicts/etc to avoid shared state across calls

**Review by dbobrenko — DISMISSED** [review]
LGTM with minor comments

File: prompting/tools/datasets/wiki.py [hunk 4]
```diff
@@ -77,13 +72,32 @@ def _get_random_titles(pages=10, seed=42) -> List:
...
+    # Fetch the HTML content of the Wikipedia article
+    url = f"https://en.wikipedia.org/wiki/{title}"
+    response = requests.get(url)
+    html_content = response.text
+
+    # Parse the HTML using BeautifulSoup
+    soup = BeautifulSoup(html_content, 'html.parser')
+
+    # Find all the sections and their content
+    sections = {}
```
[the remark this answers is not in the document]
Comment (bkb2135) [reply prompting/tools/datasets/wiki.py:89]: I just learned about the "walrus operator"

File: prompting/tools/datasets/wiki.py (hunk 3 above)
Comment (bkb2135) [reply prompting/tools/datasets/wiki.py:185]: Yeah you're totally right, this was an accidental keepsake from when I was trying a different approach

**Review by dbobrenko — APPROVED** [review]
LGTM! A few cosmetic comments that you can ignore for now.

File: tests/fixtures/dataset.py [hunk 5]
```diff
@@ -14,13 +14,17 @@
...
+    return list[0]
 
 MOCK_CONTEXT = MockDataset().next()
-WIKI_CONTEXT = WikiDataset().next(name="Emilio Alvarez (bishop)", method="get")
+WIKI_CONTEXT= WikiDataset().next(name="Emilio Alvarez (bishop)", method="get", selector=select_first)
 CODING_CONTEXT = HFCodingDataset(buffer_size=1, seed=42).next()
 MATH_CONTEXT = MathDataset(seed=123).next()
 DATEQA_CONTEXT = WikiDateDataset(seed=123).next()
 
+print(WIKI_CONTEXT)
```
Comment (dbobrenko) [inline tests/fixtures/dataset.py:26]: Is it accidentally commited?

Comment (dbobrenko) [inline tests/fixtures/dataset.py:26]:
```suggestion
```

File: tests/fixtures/dataset.py [hunk 6]
```diff
@@ -14,13 +14,17 @@
...
+
```
Comment (dbobrenko) [inline tests/fixtures/dataset.py]:
```suggestion
```

**Pull request merged**